cmd/virt-launcher/virt-launcher.go

```diff
@@ -420,6 +420,19 @@ func main() {
 		panic(err)
 	}
 
+	// This is needed for swtpm_setup, when running as the qemu user (swtpm_user = "qemu" in qemu.conf)
+	//os.Chmod("/var/lib/swtpm-localca", 0777)
+	//os.Chown("/var/lib/swtpm-localca", 107, 107)
+
+	// Manually start a swtpm daemon, even if TPM was not requested, can't hurt.
+	// TODO: add an option to only do it when TPM was requested?
+	os.MkdirAll("/tmp/tpm", 0777)
+	os.Chown("/tmp/tpm", 107,107)
+	cmd := exec.Command("/usr/bin/swtpm", "socket", "--tpmstate", "dir=/tmp/tpm", "--ctrl", "type=unixio,path=/tmp/tpm/swtpm-sock", "--tpm2", "--log", "level=20", "--daemon")
```

Comment:
Perhaps it'll be better to run this in its own container and pass this socket to the launcher? 
There are a couple of immediate benefits I can think of. For example, we can manage the memory overhead separately from the known qemu overhead. It will not mess with vCPU pinning, etc... What do you think?

Reply:
Interesting idea. While that would be nice to have, it would require a lot of work, and I'm not even sure we could handle things like migration without making changes to libvirt.
Thanks to Vasily's work, we can just leverage libvirt as it is today and rely on its full swtpm support, including the tricky migration.
I just updated the PR to do that, and I am currently testing it against Windows 11, which we really need to support as soon as possible...
If that works, it will mean that the small amount of code in this PR is all we'll have to maintain to support vTPM, which is a big win IMO.